cloud_functions/flagCryptominer.js

```diff
@@ -0,0 +1,33 @@
+const utils = require('./utils')
+
+
+const run = (mixpanel, message) => {
+  const userSubjectId = validate(message)
+  if (userSubjectId) {
+    mixpanel.people.set(`google:${userSubjectId}`, { 'isCryptominer': true }, utils.logMixpanelError)
+  }
+}
+
+const validate = message => {
+  if (!(message.attributes && message.attributes.cryptominer === true)) {
+    console.error(new Error('Unexpected message; missing `cryptominer: true` attribute. Is the Pub/Sub subscription set up correctly?'))
+    return
+  }
+  let data
+  try {
+    data = message.data && JSON.parse(Buffer.from(message.data, 'base64').toString())
+  } catch (error) {
+    console.error(utils.coerceToError(error))
```

Comment:
Assuming we want to catch errors like this, we could have a single global catch at the top-level handler, rather than having to do it at each source (and then having to guard against missing data).

Reply:
Yes. If you go back a commit, you can see my attempt at that. However, what I ran into was that the initialization of the Mixpanel SDK (which is now potentially remembered across function invocations) could fail, in which case we might need to make sure the CF instance doesn't serve any more requests; I'm not 100% sure about the semantics of `_.once` if an exception is thrown from the wrapped function.